adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRewardedAdRenderer.m

```diff
@@ -0,0 +1,133 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralRewardedAdRenderer.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+#include <stdatomic.h>
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKReward/MTGRewardAd.h>
+#import <MTGSDKReward/MTGBidRewardAdManager.h>
+
+@interface GADMAdapterMintegralRewardedAdRenderer ()<GADMediationRewardedAd,MTGRewardAdLoadDelegate,MTGRewardAdShowDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralRewardedAdRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationRewardedLoadCompletionHandler _adLoadCompletionHandler;
+    
+    /// Ad configuration for the ad to be loaded.
+    GADMediationRewardedAdConfiguration *_adConfiguration;
+
+    /// The Mintegral rewarded ad.
+    MTGBidRewardAdManager *_rewardedAd;
+    
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationRewardedAdEventDelegate> _adEventDelegate;
+}
+
+- (void)renderRewardedAdForAdConfiguration:(nonnull GADMediationRewardedAdConfiguration *)adConfiguration
+                       completionHandler:(nonnull GADMediationRewardedLoadCompletionHandler)completionHandler {
+    _adConfiguration = adConfiguration;
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationRewardedLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationRewardedAdEventDelegate>(
+                                                                        _Nullable id<GADMediationRewardedAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationRewardedAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+        
+    _rewardedAd = [MTGBidRewardAdManager sharedInstance];
```

Comment:
I would assume dot notation will also work here?
`    _rewardedAd = MTGBidRewardAdManager.sharedInstance;`

If that's the case, can you update all the access to the shared instance to use the dot notation?

Reply:
ok,i have updated.